### PR TITLE
Feature/#22 user dashboard

### PR DESCRIPTION
**Pull request by Karnich**

### CODE REVIEW

**Comment by Karnich** [comment]
closes #46

File: app/base-view/tabs/tabs.component.ts [hunk 1]
```diff
@@ -1,9 +1,21 @@
-import { Component } from "@angular/core";
+import { Component, Input, OnInit } from "@angular/core";
 
 @Component({
   selector: "rom-tabs",
   templateUrl: "base-view/tabs/tabs.component.html",
   styleUrls: ["base-view/tabs/tabs-common.css"],
 })
-export class TabsComponent {
+
```
Comment (Crevil) [inline app/base-view/tabs/tabs.component.ts]: Line break

File: app/dashboard-user/dashboard-user.component.html [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  <GridLayout rows="auto, *, auto" columns="*, auto">
+
+    <StackLayout row="0" colSpan="2" class="stats" orientation="horizontal">
+      <rom-tile text="Total records" value="10260"></rom-tile>
+      <rom-tile text="Active users" value="89"></rom-tile>
+      <rom-tile text="Total users" value="500"></rom-tile>
+    </StackLayout>
+
+    <StackLayout row="1" rowSpan="2" col="0">
```
Comment (Crevil) [inline app/dashboard-user/dashboard-user.component.html:11]: Is it necessary to wrap the chart in a layout?

File: app/dashboard-user/dashboard-user.component.html [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        <LinearAxis tkCartesianVerticalAxis></LinearAxis>
+        <BarSeries tkCartesianSeries showLabels="true" seriesName="Bar" stackMode="Stack" [items]="source" categoryProperty="Country"
+          valueProperty="Amount">
+        </BarSeries>
+      </RadCartesianChart>
+
+    </StackLayout>
+
+    <StackLayout row="2" col="1" class="buttons side-layout" orientation="vertical">
+      <StackLayout class="side-tile">
```
Comment (Crevil) [inline app/dashboard-user/dashboard-user.component.html]: Is it necessary to wrap this in a layout?

File: app/dashboard-user/dashboard-user.component.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      { Country: "Layup", Amount: 24, SecondVal: 17, ThirdVal: 23, Impact: 0, Year: 0 },
+      { Country: "Hit", Amount: 11, SecondVal: 19, ThirdVal: 24, Impact: 0, Year: 0 },
+      { Country: "Snap", Amount: 18, SecondVal: 8, ThirdVal: 21, Impact: 0, Year: 0 },
+    ];
+  }
+}
+
+// taken from nativescript-ui angular chart "getting started" guide
+// http://docs.telerik.com/devtools/nativescript-ui/Controls/Angular/Chart/getting-started
+export class Country {
```
Comment (Crevil) [inline app/dashboard-user/dashboard-user.component.ts:41]: Just a note. This is not really used for anything? Could be an interface instead.

File: app/dashboard-user/dashboard-user.module.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+import { dashboardRouting } from "./dashboard-user.routing";
+import { DashboardUserComponent } from "./dashboard-user.component";
+import { BaseViewModule } from "../base-view";
+import { TileComponent } from "./tile/tile.component";
+import { ChartDirectiveWrapper } from "../chart-directive-wrapper.module";
+
+@NgModule({
+  imports: [
+    BaseViewModule,
+    ChartDirectiveWrapper,
```
Comment (Crevil) [inline app/dashboard-user/dashboard-user.module.ts:13]: Why is this wrapper necessary?

File: app/login/login.component.ts [hunk 6]
```diff
@@ -105,8 +105,13 @@ export class LoginComponent implements OnInit {
       }
 
       if (viewer) {
-        this.router.navigate(["dashboard"]);
-        return;
+        if (!viewer.user.isConsultant && !viewer.user.isCoach) {
+          this.router.navigate(["dashboard-user"]);
+          return;
```
Comment (Crevil) [inline app/login/login.component.ts:110]: When returning here, the `else` statement is redundant. Pick one of them. ;)

Comment (Karnich) [reply app/login/login.component.ts:110]: it is.. but makes it more readable imo.. well preference i guess :p

File: app/dashboard-user/dashboard-user.module.ts (hunk 5 above)
Comment (Karnich) [reply app/dashboard-user/dashboard-user.module.ts:13]: nativescript tells me i have 2 imports of the CHART_DIRECTIVES if i import it in the NgModules that actually need them... (stacktrace) They say i need to import it in a higher level component (like app.module) OR wrap it in a module and then import that instead...  Now the import in app.modules didnt work... But this did sooo..

File: app/dashboard-user/dashboard-user.component.ts (hunk 4 above)
Comment (Karnich) [reply app/dashboard-user/dashboard-user.component.ts:41]: dummy data that should be changed at some point :p

File: app/dashboard-user/dashboard-user.component.html (hunk 3 above)
Comment (Karnich) [reply app/dashboard-user/dashboard-user.component.html]: yes. it wouldnt take the css properties of the class if i didnt. Can i wrap it in something simpler than a stacklayout if i dont need the stacklayouts functionality?

Comment (Crevil) [reply app/dashboard-user/dashboard-user.component.html]: Yeah. Use the `ContentView` class like here: https://github.com/racketometer/frontend-application/pull/46/files#diff-5921e875726fa21c5f8911e700c4067bR7

File: app/dashboard-user/dashboard-user.module.ts (hunk 5 above)
Comment (Crevil) [reply app/dashboard-user/dashboard-user.module.ts:13]: All right. Makes sence. ;)

**Pull request merged**